pkg/controller/restore_controller.go

```diff
@@ -280,6 +284,11 @@ func (controller *restoreController) processRestore(key string) error {
 	for _, e := range restoreErrors.Namespaces {
 		restore.Status.Errors += len(e)
 	}
+	if restore.Status.Errors > 0 {
```

Comment:
xref https://github.com/heptio/ark/pull/531#discussion_r195196289 cc @ashish-amarnath @rosskukulinski - this is 1 place where we currently have phases `New`, `InProgress`, `FailedValidation`, and `Completed`. Once the restore is `Completed`, it has 0+ warnings and 0+ errors. How do you think we should record this information as metrics?

Reply:
On the backup side, when they fail validation I don't think they even get counted as starting if they failed validation. The attempts are logged at [L277](https://github.com/heptio/ark/blob/master/pkg/controller/backup_controller.go#L277), but the validation failure returns at [L267](https://github.com/heptio/ark/blob/master/pkg/controller/backup_controller.go#L267).

Do we want to register attempts for restores/backups before doing the validation?

Reply:
> Do we want to register attempts for restores/backups before doing the validation?

+1. We may want to add a metric for `backup/restore_validation_failure_total`?

Reply:
We probably should have gauges for each of phases New, InProgress with the current number of backups (and also restores) in those states.  That will help alert users if they have backups that are stuck for some reason.

As for 0+ warnings and 0+ errors, we should increment a counter each (per-schedule, or empty schedule) (plus the counter somewhere for backups completed).  Users won't be able to know _which_ backup/restore had Errors or Warnings, but they will know that it happened and can then go look at Ark server logs (or describe backup/restore).

Reply:
> Maybe we can say backup completed with %d errors and %d warnings and track the error and warning counts inside the backup object?

We do plan to do this, similar to how we record this information for restores, but we haven't done it yet.

> Validation to me reads like there is something wrong with the way the backup object was created.

Restore validation errors include:
- restore spec is missing backup name
- ark couldn't find the backup you want to restore
- you asked to include a non-restorable resource in the restore (nodes, events, backups, restores)
- the requested include/exclude namespaces overlap
- the requested include/exclude resources overlap
- you asked to restore PVs but ark isn't configured with a PV provider